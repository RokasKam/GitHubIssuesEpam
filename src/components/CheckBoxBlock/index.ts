import { CheckBoxBlock } from "./CheckBoxBlock";
export { CheckBoxBlock };