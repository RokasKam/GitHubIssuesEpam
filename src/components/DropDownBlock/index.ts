import { DropDownBlock } from "./DropDownBlock";
export { DropDownBlock };