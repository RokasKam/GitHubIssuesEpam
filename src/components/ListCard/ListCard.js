import React from 'react'
import { Text, View } from 'react-native';
import { Card } from 'react-native-elements';
import { styles } from './ListCard.style';

export const ListCard = ({ itemId, itemTitle }) => {
    return (
        <Card containerStyle={styles.cardStyle}>
            <View>
                <Text style={styles.cardTextStyle}>
                    {itemTitle}
                </Text>
            </View>
            <View >
                <Text style={styles.cardTextStyle}>
                    {itemId}
                </Text>
            </View>
        </Card>
    )
};
