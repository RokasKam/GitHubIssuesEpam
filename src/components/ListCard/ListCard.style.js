import { StyleSheet, Dimensions } from 'react-native';

export const styles = StyleSheet.create({
    cardStyle: {
        color: "white",
        height: Dimensions.get("screen").height / 7,
        width: Dimensions.get("window").width / 1.2,
        borderRadius: 15,
        padding: 15,
        marginTop: 15,
        marginLeft: 8,
        marginBottom: 8,
        marginRight: 8,
    },
    cardTextStyle: {
        fontFamily: "Hiragino sans",
        marginBottom: 10,
        fontStyle: "normal",
        fontWeight: "500",
        fontSize: Dimensions.get("screen").height / 60,
    },
});