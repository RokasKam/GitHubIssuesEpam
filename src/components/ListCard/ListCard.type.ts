export type Props = {
    itemId: string;
    itemTitle: string;
};