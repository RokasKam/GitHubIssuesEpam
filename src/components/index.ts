export { CheckBoxBlock } from "./CheckBoxBlock";
export { DropDownBlock } from "./DropDownBlock";
export { IssuesList } from "./IssuesList";
export { ListCard } from "./ListCard";
export { SearchButton } from "./SearchButton";
export { SearchInput } from "./SearchInput";

