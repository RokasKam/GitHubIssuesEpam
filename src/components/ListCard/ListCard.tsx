import React from 'react'
import { Text, View } from 'react-native';
import { Card } from 'react-native-elements';
import { styles } from './ListCard.style';
import { Props } from './ListCard.type';

export const ListCard: React.FC<Props> = ({ itemId, itemTitle }) => {
    return (
        //@ts-ignore
        <Card containerStyle={styles.cardStyle}>
            <View>
                <Text style={styles.cardTextStyle}>
                    {itemTitle}
                </Text>
            </View>
            <View >
                <Text style={styles.cardTextStyle}>
                    {itemId}
                </Text>
            </View>
        </Card>
    )
};
