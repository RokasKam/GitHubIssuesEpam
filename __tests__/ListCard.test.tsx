import React from 'react';
import renderer from 'react-test-renderer';
import { ListCard } from '../src/components/ListCard'

test('renders correctly', () => {
  const tree = renderer.create(<ListCard itemId={''} itemTitle={''} />).toJSON();
  expect(tree).toMatchSnapshot();
});