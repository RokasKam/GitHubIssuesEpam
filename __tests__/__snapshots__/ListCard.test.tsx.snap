// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "borderColor": "#e1e8ee",
      "borderRadius": 15,
      "borderWidth": 1,
      "color": "white",
      "height": 190.57142857142858,
      "margin": 15,
      "marginBottom": 8,
      "marginLeft": 8,
      "marginRight": 8,
      "marginTop": 15,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
      "width": 625,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View>
      <Text
        style={
          Object {
            "fontFamily": "Hiragino sans",
            "fontSize": 22.233333333333334,
            "fontStyle": "normal",
            "fontWeight": "500",
            "marginBottom": 10,
          }
        }
      />
    </View>
    <View>
      <Text
        style={
          Object {
            "fontFamily": "Hiragino sans",
            "fontSize": 22.233333333333334,
            "fontStyle": "normal",
            "fontWeight": "500",
            "marginBottom": 10,
          }
        }
      />
    </View>
  </View>
</View>
`;
